src/managers/AuditLogManager.ts

```diff
@@ -0,0 +1,97 @@
+import { EventEmitter } from '@tauri-apps/api/shell';
+
+export type RequestEvent = {
+	timestamp: Date;
+	chronology: 'before' | 'after';
+	eventType: `${'pre' | 'post'}${'Service' | 'Endpoint' | 'Request'}Script` | 'request' | 'root';
```

Comment:
I feel like this might be more performant if it were a nested set of properties, given you reference elements of it elsewhere (like with string.includes pre/post)?

Reply:
Can you give an example of how you'd have it look like? I think this is a good idea

Reply:
Now that I'm thinking about it more I've realized the implications of the `| 'request' | 'root'` and that makes it a lot more difficult to split this out, so I think it's fine - not ideal, but other options might be worse.

Reply:
Alright I didn't read the below comment. I think given it's causing weird stuff like "Request | request" the solution might not be worse. Let me think on it for a bit and I'll get back to you by end of day.